main-settings/src/main/scala/sbt/Def.scala

```diff
@@ -132,13 +133,19 @@ object Def extends Init[Scope] with TaskMacroExtra {
       project: Reference,
       trailingSlash: Boolean
   ): String = {
-    val trailing = if (trailingSlash) " /" else ""
-    project match {
-      case BuildRef(current.build)      => "ThisBuild" + trailing
-      case `current`                    => ""
-      case ProjectRef(current.build, x) => x + trailing
-      case _                            => Reference.display(project) + trailing
+    import Reference.{ display => displayRef }
+    @tailrec def loop(ref: Reference): String = ref match {
+      case ProjectRef(b, p) => if (b == current.build) loop(LocalProject(p)) else displayRef(ref)
+      case BuildRef(b)      => if (b == current.build) loop(ThisBuild) else displayRef(ref)
+      case RootProject(b)   => if (b == current.build) loop(LocalRootProject) else displayRef(ref)
+      case LocalProject(p)  => if (p == current.project) "" else p
+      case ThisBuild        => "ThisBuild"
+      case LocalRootProject => "<root>"
+      case ThisProject      => "<this>"
     }
+    val str = loop(project)
+    if (trailingSlash && !str.isEmpty) s"$str /"
+    else str
```

Comment:
What are the downstream of `displayRelative`? i.e. how do they show up in the UI?

Reply:
I think I hit it through `Extracted` when running a task out of graph (in an early version of scala/scala#8662, before I switch to using `set`) where I used `LocalProject` and got the `{<this>}bar / javaOptions` output rather than just `bar / javaOptions`.